### PR TITLE
Add Zip<C: Collection> operator to Single trait

### PR DESCRIPTION
**Pull request by yuzushioh**
I added `fund zip<C: Collection>(_ collection: C, resultSelector: _)` operator to Single trait.

### CODE REVIEW

File: Tests/RxSwiftTests/PrimitiveSequenceTest+zip+arity.swift [hunk 1]
```diff
@@ -11,6 +11,29 @@ import XCTest
...
+    func testZipCollection_tuple() {
+        let collection = [Single<Int>.just(1), Single<Int>.just(1), Single<Int>.just(1)]
+        let singleResult: Single<Int> = Single.zip(collection).map { $0.reduce(0, combine: +) }
+        
+        let result = try! singleResult
+            .toBlocking().first()!
+        
+        XCTAssertEqual(result, 3)
+    }
+}
```
Comment (kzaher) [inline Tests/RxSwiftTests/PrimitiveSequenceTest+zip+arity.swift]: Could you please move contents of these tests to `SingleTest`.

File: RxSwift/Traits/PrimitiveSequence+Zip+Collection.swift [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    /**
+     Merges the specified observable sequences into one observable sequence all of the observable sequences have produced an element at a corresponding index.
+     
+     - returns: An observable sequence containing the result of combining elements of the sources.
+     */
+    public static func zip<C: Collection>(_ collection: C) -> PrimitiveSequence<TraitType, [ElementType]> where C.Iterator.Element == PrimitiveSequence<TraitType, ElementType> {
+        let raw = Observable.zip(collection.map { $0.asObservable() })
+        return PrimitiveSequence(raw: raw)
+    }
+}
```
Comment (kzaher) [inline RxSwift/Traits/PrimitiveSequence+Zip+Collection.swift]: Could you please move these extensions to `Single.swift`.

File: RxSwift/Traits/PrimitiveSequence+Zip+Collection.swift [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+// MARK: zip + collection
+
+extension PrimitiveSequenceType where TraitType == SingleTrait {
+    /**
+     Merges the specified observable sequences into one observable sequence by using the selector function whenever all of the observable sequences have produced an element at a corresponding index.
+     
+     - parameter resultSelector: Function to invoke for each series of elements at corresponding indexes in the sources.
+     - returns: An observable sequence containing the result of combining elements of the sources using the specified result selector function.
+     */
+    public static func zip<C: Collection, R>(_ collection: C, _ resultSelector: @escaping ([ElementType]) throws -> R) -> PrimitiveSequence<TraitType, R> where C.Iterator.Element == PrimitiveSequence<TraitType, ElementType> {
```
Comment (kzaher) [inline RxSwift/Traits/PrimitiveSequence+Zip+Collection.swift]: There is one additional condition that needs to be checked here.

```swift
if collection.isEmpty {
       return PrimitiveSequence<TraitType, R>(raw: .just(resultSelector([])))
}
```

And unit test added.

File: RxSwift/Traits/PrimitiveSequence+Zip+Collection.swift [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        let raw = Observable.zip(collection.map { $0.asObservable() }, resultSelector)
+        return PrimitiveSequence<TraitType, R>(raw: raw)
+    }
+    
+    /**
+     Merges the specified observable sequences into one observable sequence all of the observable sequences have produced an element at a corresponding index.
+     
+     - returns: An observable sequence containing the result of combining elements of the sources.
+     */
+    public static func zip<C: Collection>(_ collection: C) -> PrimitiveSequence<TraitType, [ElementType]> where C.Iterator.Element == PrimitiveSequence<TraitType, ElementType> {
```
Comment (kzaher) [inline RxSwift/Traits/PrimitiveSequence+Zip+Collection.swift]: Equivalent here.

**Comment by yuzushioh** [comment]
I resubmitted it another PR

**Pull request closed without merging**